src/dialogs/vkbasalt.py

```diff
@@ -0,0 +1,265 @@
+                VkBasaltSettings.dls_sharpness = Gtk.Adjustment.get_value(self.dls_sharpness)
+                VkBasaltSettings.dls_denoise = Gtk.Adjustment.get_value(self.dls_denoise)
+            if self.fxaa.get_enable_expansion() is True:
+                effects.append("fxaa")
+                VkBasaltSettings.fxaa_subpixel_quality = Gtk.Adjustment.get_value(self.fxaa_subpixel_quality)
+                VkBasaltSettings.fxaa_quality_edge_threshold = Gtk.Adjustment.get_value(self.fxaa_quality_edge_threshold)
+                VkBasaltSettings.fxaa_quality_edge_threshold_min = Gtk.Adjustment.get_value(self.fxaa_quality_edge_threshold_min)
+            if self.smaa.get_enable_expansion() is True:
+                effects.append("smaa")
+                VkBasaltSettings.smaa_threshold = Gtk.Adjustment.get_value(self.smaa_threshold)
+                VkBasaltSettings.smaa_edge_detection = self.smaa_edge_detection
+                VkBasaltSettings.smaa_corner_rounding = Gtk.Adjustment.get_value(self.smaa_corner_rounding)
+                VkBasaltSettings.smaa_max_search_steps = Gtk.Adjustment.get_value(self.smaa_max_search_steps)
+                VkBasaltSettings.smaa_max_search_steps_diagonal = Gtk.Adjustment.get_value(self.smaa_max_search_steps_diagonal)
+            if self.clut.get_enable_expansion() is True:
+                VkBasaltSettings.lut_file_path = self.lut_file_path
+
+            VkBasaltSettings.disable_on_launch = self.switch_disable_on_launch.get_state()
+
+        VkBasaltSettings.effects = tuple(effects)
+
+        VkBasaltSettings.output = conf
+
+        parse(VkBasaltSettings)
+        self.close()
+
+    def __save(self, *args):
+        GLib.idle_add(self.__idle_save)
+
+    def __default(self, widget, state):
+        self.cas.set_sensitive(not state)
+        self.dls.set_sensitive(not state)
+        self.fxaa.set_sensitive(not state)
+        self.smaa.set_sensitive(not state)
+        self.row_disable_on_launch.set_sensitive(not state)
+        self.clut.set_sensitive(not state)
+
+    def __change_edge_detection_type(self, widget, edge_detection_type):
+        self.smaa_edge_detection = edge_detection_type
+        self.luma.handler_block_by_func(self.__change_edge_detection_type)
+        self.color.handler_block_by_func(self.__change_edge_detection_type)
+        if edge_detection_type == "luma":
+            self.color.set_active(False)
+            self.luma.set_active(True)
+        elif edge_detection_type == "color":
+            self.color.set_active(True)
+            self.luma.set_active(False)
+
+        self.luma.handler_unblock_by_func(self.__change_edge_detection_type)
+        self.color.handler_unblock_by_func(self.__change_edge_detection_type)
+
+    def __import_clut(self, *args):
+        def set_path(_dialog, response, _file_dialog):
+            if response == -3:
+                self.lut_file_path = _file_dialog.get_file().get_path()
+
+                img = Image.open(self.lut_file_path)
```

Comment:
I may be wrong but this should be achievable using GdkPixbuf too

Reply:
Correct.

Reply:
Use Gdk.Texture.new_from_filename¹ or new_from_file. Pixbufs are old gtk3 api. Textures are a subclass of Paintables, GtkImage has a `.set_paintable` method.

¹https://docs.gtk.org/gdk4/ctor.Texture.new_from_filename.html

Reply:
I'm only using it to get the width and height of the image.

Reply:
There are methods for paintable to get those, see https://docs.gtk.org/gdk4/vfunc.Paintable.get_intrinsic_height.html

Reply:
Thanks.